privacyidea/lib/resolvers/HTTPResolver.py

```diff
@@ -0,0 +1,248 @@
+        return self.config['endpoint'] if 'endpoint' in self.config else ''
+
+    def loadConfig(self, config):
+        """
+        Load the configuration from the dict into the Resolver object.
+        If attributes are missing, need to set default values.
+        If required attributes are missing, this should raise an
+        Exception.
+
+        :param config: The configuration values of the resolver
+        :type config: dict
+        """
+
+        # Validate method
+        method = config.get('method')
+        if not method or method.lower() not in ('get', 'post'):
+            raise Exception('Validation Error: "method" must be "get" or "post"')
+
+        # Validate endpoint
+        if not config.get('endpoint'):
+            raise Exception('Validation Error: "endpoint" must be set')
+
+        if not config.get('requestMapping'):
+            raise Exception('Validation Error: "request mapping" input is required')
+
+        if not config.get('responseMapping'):
+            raise Exception('Validation Error: "response mapping" input is required')
+
+        # Validate special error handler
+        if config.get('hasSpecialErrorHandler') and not config.get('errorResponse'):
+            raise Exception('Validation Error: "error response" input must be set if you enable special error handler')
+
+        self.config = config
+        return self
+
+    @classmethod
+    def testconnection(cls, param):
+        """
+        This function lets you test if the parameters can be used to create a
+        working resolver. Also, you can use it anytime you see if the API is
+        running as expected.
+        The implementation should try to make a request to the HTTP API and verify
+        if user can be retrieved.
+        In case of success it should return the raw http response.
+
+        :param param: The parameters that should be saved as the resolver
+        :type param: dict
+        :return: returns True in case of success and a raw response
+        :rtype: tuple
+        """
+        desc = ""
+        success = False
+        try:
+            resolver = HTTPResolver()
+            resolver.loadConfig(param)
+            response = resolver._getUser(param.get('testUser'))
+            desc = response
+            success = True
+        except Exception as e:
+            success = False
+            desc = "failed: {0!s}".format(e)
+        return success, desc
+
+    #
+    #   Private methods
+    #
+    def _getUser(self, userid):
+        param = self.config
+        method = param.get('method').lower()
+        endpoint = param.get('endpoint')
+        requestMappingJSON = json.loads(param.get('requestMapping').replace("{userid}", userid))
+        responseMapping = json.loads(param.get('responseMapping'))
+        headers = json.loads(param.get('headers', '{}'))
+        hasSpecialErrorHandler = bool(param.get('hasSpecialErrorHandler'))
+        errorResponse = json.loads(param.get('errorResponse'))
```

Comment:
Now this would throw a `JSONDecodeError` if the parameter is empty. I think we should leave it as before (with a default value of `'{}'`= and check below if `errorResponse` is `True`  (an empty dict evaluates to `False`).

Reply:
Since this value can still be empty i would use
`errorResponse = json.loads(param.get('errorResponse', '{}'))`
to avoid the `JSONDecodeError`.

Reply:
I didn't add the default value because it's validated at the ` getConfig() step. But if I remove validations according to [this](https://github.com/privacyidea/privacyidea/pull/2083#discussion_r469853298) I definitely have to add it.

Reply:
Yes, that would be the idea since this entry is optional, it can be empty in the db (or we check it together with the `hasSpecialErrorHandler` option, but i think, just using the default string here should be enough).